packages/application-shell/src/components/user-settings-menu/user-settings-menu.js

```diff
@@ -18,39 +18,36 @@ import {
   SUPPORT_PORTAL_URL,
 } from '@commercetools-frontend/constants';
 import withApplicationsMenu from '../with-applications-menu';
-import withMouseOverState from '../with-mouse-over-state';
 import handleApolloErrors from '../handle-apollo-errors';
 import styles from './user-settings-menu.mod.css';
 import messages from './messages';
 
-const UserAvatar = props => (
-  <div onMouseOver={props.handleMouseOver} onMouseOut={props.handleMouseOut}>
-    <Spacings.Inline alignItems="center">
-      <Avatar
-        gravatarHash={props.gravatarHash}
-        firstName={props.firstName}
-        lastName={props.lastName}
-        isHighlighted={props.isMouseOver}
-      />
-      <CaretDownIcon
-        size="small"
-        theme={props.isMouseOver ? 'grey' : 'black'}
-      />
-    </Spacings.Inline>
-  </div>
-);
+const UserAvatar = props => {
+  const [isMouseOver, setMouseOver] = React.useState(false);
+  return (
+    <div
+      onMouseOver={() => setMouseOver(true)}
```

Comment:
Nit: we could define to setters: setMouseOver and setMouseOut and use React.useCallback to keep the fn reference equal.